plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/ShellUtils.java

```diff
@@ -40,6 +43,35 @@ public static boolean launchProgram(@NotNull String path) {
         return Program.launch(path);
     }
 
+    /**
+     * Opens a file under the given {@code path} using preferred for the current platform application.
+     *
+     * @param path path to a file to open
+     * @return {@code true} on success, {@code false} on failure if the file can't be opened
+     */
+    public static boolean openExternalFile(@NotNull Path path) {
+        try {
+            if (RuntimeUtils.isMacOS()) {
+                executeWithReturnCodeCheck("open", "-a", "Finder.app", path.toAbsolutePath().toString());
+                return true;
+            } else if (RuntimeUtils.isLinux()) {
+                executeWithReturnCodeCheck("xdg-open", path.toAbsolutePath().toString());
+                return true;
+            }
+        } catch (IOException | InterruptedException e) {
+            log.debug("Unable to open external program in a platform-specific way: " + e.getMessage());
+            return false;
+        }
+
+        try {
+            Desktop.getDesktop().open(path.toFile());
+            return true;
+        } catch (IOException e) {
```

Comment:
Linux may not have 'xdg-open' in some cases, we should use try to use 'getDesktop().open()' in that case.

Reply:
That `return false;` was excessive, I agree.